state/txindex/kv/kv.go

```diff
@@ -154,10 +159,7 @@ func (txi *TxIndex) Search(q *query.Query) ([]*types.TxResult, error) {
 	skipIndexes := make([]int, 0)
 
 	// if there is a height condition ("tx.height=3"), extract it for faster lookups
-	height, heightIndex := lookForHeight(conditions)
-	if heightIndex >= 0 {
-		skipIndexes = append(skipIndexes, heightIndex)
-	}
+	height, _ := lookForHeight(conditions)
```

Comment:
Looks like we don't need heightIndex anymore. Can you modify `lookForHeight` to only return height then?

Reply:
@melekes sure, thanks. Will refactor in next commit